src/xmake.ts

```diff
@@ -177,6 +177,11 @@ export class XMake implements vscode.Disposable {
         log.verbose("updating Diagnosis ..");
         const result = await process.runv(config.executable, ["check"], { "COLORTERM": "nocolor" }, config.workingDirectory);
         const diags = diagnosis.parse(result.stdout ?? result.stderr);
+        // clear if no errors
+        if (Object.keys(diags).length === 0) {
```

Comment:
it might be better to clear diagnosis all the times?

since waruqi has made a change to check it from root every time, it should generate all errors/warnings. better to remove all existing ones first

see commit: https://github.com/xmake-io/xmake-vscode/commit/09e26c04436fec707432b08d7472395af015aea6